plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcDeletedRows.java

```diff
@@ -324,9 +320,9 @@ public Optional<Set<RowId>> loadOrYield()
                     if (currentPageSource == null) {
                         AcidInfo.DeleteDeltaInfo deleteDeltaInfo = deleteDeltas.next();
                         currentPath = createPath(acidInfo, deleteDeltaInfo, sourceFileName);
-                        FileSystem fileSystem = hdfsEnvironment.getFileSystem(identity, currentPath, configuration);
-                        FileStatus fileStatus = hdfsEnvironment.doAs(identity, () -> fileSystem.getFileStatus(currentPath));
-                        currentPageSource = pageSourceFactory.createPageSource(fileStatus.getPath().toString(), fileStatus.getLen()).orElseGet(() -> new EmptyPageSource());
+                        TrinoFileSystem fileSystem = fileSystemFactory.create(identity);
+                        TrinoInputFile inputFile = fileSystem.newInputFile(currentPath.toString());
+                        currentPageSource = pageSourceFactory.createPageSource(inputFile.location(), inputFile.length()).orElseGet(() -> new EmptyPageSource());
```

Comment:
Is `location` equvalent to `fileStatus.getPath`?

Reply:
In `HdfsInputFile` implementation, which is only `non test only`  implementation, it returns path to the file. 
Same as `FileStatus.getPath()`:
```
public String getFilePath()
    {
        return filePath;
    }
```

Reply:
`io.trino.plugin.hive.s3.TrinoS3FileSystem#getFileStatus` uses `io.trino.plugin.hive.s3.TrinoS3FileSystem#qualifiedPath` for `FileStatus` path. Is this same as `location`?

Reply:
Oh right, I checked wrong `FileStatus` class.
With the current TrinoFileSystem api I don't see any other way to do it.. 

But I think we are fine with current implementation:
- this location is later used by another instance of TrinoFileSystem in the exactly same way - creating new InputFile from that path
- `location` will return exactly the same thing that was passed to `fileSystem.newInputFile` so `currentPath.toString()`. It will have the same result as previous implementation as in the end `toString()` was called on the provided instance of `Path` here `pageSourceFactory.createPageSource(fileStatus.getPath().toString(),...)`

Reply:
> was called on the provided instance of Path here pageSourceFactory.createPageSource(fileStatus.getPath().toString(),...)

But isn't file `fileStatus.getPath()` e.g. absolute or expanded location?

> location will return exactly the same thing that was passed to fileSystem.newInputFile so currentPath.toString()

I don't think it's true, as `io.trino.filesystem.hdfs.HadoopPaths#hadoopPath` is called in `HdfsInputFile` constructor

Reply:
So FileStatus in its constructor accepts path of type `org.apache.hadoop.fs.Path`, it doesn't do anything to it, then in the previous version of code it returned it and called `toString()` on it.

Current version does following things:
- gets the same path and calls `toString()` on it
- it calls `hadoopPath()` on it and stores it in `private final Path file` field it also stores original path in `private final String path`  field.
- we call location on that inputFile and location returns path field so invocation of `hadoopPath` is not important here

so the only difference is that previous version called toString at the end  and current calls it at the beginning

Reply:
Discussed offline